app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreen.kt

```diff
@@ -0,0 +1,300 @@
+        horizontalArrangement = Arrangement.spacedBy(MaterialTheme.spacing.large),
+        userScrollEnabled = false,
+        modifier = Modifier
+            .padding(dimensionResource(id = R.dimen.medium_spacing))
+    ) {
+        items(upcomingViewState.movieCategories.size) { category ->
+            MovieCategoryLabel(
+                movieCategoryLabelViewState = MovieCategoryLabelViewState(
+                    upcomingViewState.movieCategories[category].itemId,
+                    upcomingViewState.movieCategories[category].isSelected,
+                    upcomingViewState.movieCategories[category].categoryText
+                ),
+                onClick = { onLabelClick(upcomingViewState.movieCategories[category].itemId) },
+                modifier = Modifier
+            )
+        }
+    }
+
+    LazyRow(
+        horizontalArrangement = Arrangement.spacedBy(MaterialTheme.spacing.small),
+        modifier = Modifier
+            .padding(dimensionResource(id = R.dimen.medium_spacing))
+    ) {
+        items(upcomingViewState.movies.size) { movie ->
+            MovieCard(
+                movieCardViewState = MovieCardViewState(
+                    imageUrl = upcomingViewState.movies[movie].imageUrl,
+                    id = upcomingViewState.movies[movie].movieId,
+                    isFavorite = upcomingViewState.movies[movie].isFavorite,
+                    title = upcomingViewState.movies[movie].title
+                ),
+                onClick = { onNavigateToMovieDetails(upcomingViewState.movies[movie].movieId) },
+                onFavoriteButtonClicked = {},
+                modifier = Modifier
+            )
+        }
+    }
+}
+
+@Composable
+fun NowPlayingMovies(
+    nowPlayingViewState: HomeMovieCategoryViewState,
+    onLabelClick: (Int) -> Unit,
+    onNavigateToMovieDetails: (Int) -> Unit
+) {
+    Text(
+        text = stringResource(id = R.string.now_playing),
+        fontSize = 17.sp,
+        fontWeight = FontWeight.Bold,
+        color = MaterialTheme.colors.primary,
+        modifier = Modifier
+            .padding(MaterialTheme.spacing.small)
+    )
+
+    LazyRow(
+        horizontalArrangement = Arrangement.spacedBy(MaterialTheme.spacing.large),
+        userScrollEnabled = false,
+        modifier = Modifier
+            .padding(dimensionResource(id = R.dimen.medium_spacing))
+    ) {
+        items(nowPlayingViewState.movieCategories.size) { category ->
+            MovieCategoryLabel(
+                movieCategoryLabelViewState = MovieCategoryLabelViewState(
+                    nowPlayingViewState.movieCategories[category].itemId,
+                    nowPlayingViewState.movieCategories[category].isSelected,
+                    nowPlayingViewState.movieCategories[category].categoryText
+                ),
+                onClick = { onLabelClick(nowPlayingViewState.movieCategories[category].itemId) },
+                modifier = Modifier
+            )
+        }
+    }
+
+    LazyRow(
```

Comment:
you're creating this many times and all are the same except list of items - please create Composable component and reuse it